src/modules/checkIsValidUserInput.js

```
@@ -9,7 +9,25 @@
  * @return {boolean} - True if the user input is valid, false otherwise
  */
 function checkIsValidUserInput(userInput) {
-  /* Write your code here */
+  if (userInput[0] === '0') {
+    return false;
+  }
+
+  if (userInput.length !== 4) {
+    return false;
+  }
+
+  if (isNaN(+userInput)) {
+    return false;
+  }
+
+  const digits = new Set([...userInput]);
+
+  if (digits.size !== 4) {
+    return false;
+  }
+
+  return true;
```

Issue: The function returns true if the input is valid, but according to the requirements, it should not return a boolean value. It should throw errors for invalid input and do nothing (return undefined) if the input is valid.